Library/Homebrew/cmd/update.sh

```diff
@@ -390,6 +390,22 @@ EOS
     fi
   fi
 
+  if [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core/.git/shallow" ]]
+  then
+    odie <<EOS
+homebrew-core is a shallow clone. To \`brew update\` first run:
+  git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" fetch --unshallow
+EOS
+  fi
+
+  if [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask/.git/shallow" ]]
+  then
+    odie <<EOS
+homebrew-cask is a shallow clone. To \`brew update\` first run:
+  git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask" fetch --unshallow
+EOS
+  fi
+
```

Comment:
It might be useful to add a comment we're doing this per GitHub's request, so that future maintainers know why.
 
(Not sure if this would be an improvement but...)
We can convert these two blocks to a single 'for' loop:
```sh
  for repo in core cask; do
    if [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-$repo/.git/shallow" ]]
    then
      odie <<EOS
homebrew-$repo is a shallow clone. To \`brew update\`, first run:
      git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-$repo" fetch --unshallow
EOS
    fi
  done
```
or even a loop across all taps:
```sh
  for dir in "$HOMEBREW_LIBRARY"/Taps/*/*; do
    if [[ -f "$dir/.git/shallow" ]]
    then
      odie <<EOS
${dir##$HOMEBREW_LIBRARY/Taps/} is a shallow clone. To \`brew update\`, first run:
      git -C "$dir" fetch --unshallow
EOS
    fi
  done
```

I might be mistaken but I think there should be a comma before `first run`.

Reply:
> It might be useful to add a comment that we're doing this per GitHub's request, so that future maintainers know why.

Good call, added.

Passing on the loop due to https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)